lib/pbench/server/auth/__init__.py

```diff
@@ -62,6 +61,8 @@ def __init__(
         self.headers = CaseInsensitiveDict([] if headers is None else headers)
         self.verify = verify
         self.connection = requests.session()
+        self.public_key = self._get_oidc_public_key()
+        # TODO: Think about a better way to set well known endpoints
```

Comment:
The `TODO` could reference the issue, and also might clarify that the relative positioning of the call in the constructor is important as it depends on instance attributes.